src/main/kotlin/MaskinportenClientConfig.kt

```diff
@@ -7,17 +7,106 @@ import com.nimbusds.jose.crypto.RSASSASigner
 import com.nimbusds.jose.jwk.RSAKey
 import com.nimbusds.jwt.JWTClaimsSet
 import com.nimbusds.jwt.SignedJWT
+import java.security.KeyFactory
+import java.security.PrivateKey
+import java.security.spec.PKCS8EncodedKeySpec
 import java.time.Instant
+import java.util.Base64
 import java.util.Date
 import java.util.UUID
 
-data class MaskinportenClientConfig(
-    val scope: String,
-    val clientId: String,
-    val clientJwk: String,
-    val issuer: String,
+interface MaskinportenClientConfig {
+    val scope: String
+    val clientId: String
+    val issuer: String
     val endpoint: String
-) {
+    fun getJwtAssertion(): String
+}
+
+/**
+ * MaskinportenClientConfigPkey er en implementasjon av MaskinportenClientConfig med privatekey som autentiseringsmetode for maskinporten
+ *
+ * @param kid  Det er id-en til Nøkkelen key-id (kid)
+ * @param privateKey  Det er privatekey som skal brukes til å signere JWT tokenet
+ * @param clientId  issuer - Din egen client_id.
+ * @param issuer Audience - issuer-identifikatoren til Maskinporten. Verdi for aktuelt miljø finner du på .well-known-endpunkt.
+ * @param consumerOrgNr  Det er organisasjonsnummeret til virksomheten som skal bruke maskinporten på vegne av
+ * @param scope Space-separert liste over scopes som klienten forespør.
+ * @param endpoint  Det er endepunktet til maskinporten
+ *
+ */
+class MaskinportenClientConfigPkey(
+    val kid: String,
```

Comment:
Stemmer for `keyId`, så det blir enda tydeligere der man skal bruke klienten hva dette er.